### PR TITLE
fix: display delegate information in summary step for vote txs

### PR DESCRIPTION
**Pull request by goga-m**
## Summary
Closes https://app.clickup.com/t/86dux5j0y

## Checklist

-   [ ] My changes look good in both light AND dark mode
-   [ ] The change is not hardcoded to a single network, but has multi-asset in mind
-   [ ] I checked my changes for obvious issues, debug statements and commented code
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

**Review by samharperpittam — APPROVED** [review]

File: src/domains/transaction/hooks/use-transaction-voting-wallets.ts [hunk 1]
```diff
@@ -1,14 +1,31 @@
...
+ */
+const extractVotingData = ({ transaction }: { transaction: DTO.RawTransactionData }) => {
+	const data =
+		typeof transaction.data?.().data === "function" ? transaction.data?.().data() : transaction.data?.().data;
+	const votes = data?.asset?.votes ?? [];
+	const unvotes = data?.asset?.unvotes ?? [];
+
+	return {
+		unvotes: unvotes.map((publikKey: string) => publikKey.replace(/^[+-]+/, "")),
+		votes: votes.map((publikKey: string) => publikKey.replace(/^[+-]+/, "")),
```
Comment (ItsANameToo) [inline src/domains/transaction/hooks/use-transaction-voting-wallets.ts:25]:
```suggestion
		unvotes: unvotes.map((publicKey: string) => publicKey.replace(/^[+-]+/, "")),
		votes: votes.map((publicKey: string) => publicKey.replace(/^[+-]+/, "")),
```

File: src/domains/transaction/hooks/use-transaction-voting-wallets.ts [hunk 2]
```diff
@@ -1,14 +1,31 @@
...
+/**
+ * @TODO: This can be handled in SDK to retrieve voting and unvoting public keys (or generally the asset data)
+ * through a common method/format, whether it's a signed or confirmed transaction.
+ * Currently `data` is a function in signed transaction and an object in confirmed transaction.
+ */
+const extractVotingData = ({ transaction }: { transaction: DTO.RawTransactionData }) => {
+	const data =
+		typeof transaction.data?.().data === "function" ? transaction.data?.().data() : transaction.data?.().data;
+	const votes = data?.asset?.votes ?? [];
+	const unvotes = data?.asset?.unvotes ?? [];
```
Comment (ItsANameToo) [inline src/domains/transaction/hooks/use-transaction-voting-wallets.ts:21]: maybe turn this into two helpers, one for confirmed (using the `votes` and `unvotes` methods) and one for the signed tx that uses the raw asset? that will make it easier to follow why these checks are done and different data is accessed

File: src/domains/transaction/hooks/use-transaction-voting-wallets.ts [hunk 3]
```diff
@@ -1,14 +1,31 @@
...
 
 interface Properties {
 	network: Networks.Network;
-	transaction: DTO.ExtendedConfirmedTransactionData;
+	transaction: DTO.RawTransactionData;
 	profile: Contracts.IProfile;
 }
 
+/**
+ * @TODO: This can be handled in SDK to retrieve voting and unvoting public keys (or generally the asset data)
```
Comment (ItsANameToo) [inline src/domains/transaction/hooks/use-transaction-voting-wallets.ts:13]: can you add a discussion card for this to clickup please

**Review by ItsANameToo — CHANGES_REQUESTED** [review]
some comments and failing tests @goga-m

**Review by ItsANameToo — APPROVED** [review]

**Comment by ItsANameToo** [comment]
also https://app.clickup.com/t/86duyv9rd

**Pull request merged**